src/sentry/sentry_metrics/querying/metadata/utils.py

```diff
@@ -104,6 +106,40 @@ def _transform_conditions_with(condition: Condition) -> Optional[ConditionGroup]
     return _visit_conditions(conditions, _transform_conditions_with)
 
 
+def transform_latest_release_condition(
```

Comment:
This was implemented also here to avoid making the PR too complex. During refactor fridays I will take care of removing the duplication.